### DISCUSSION TITLE
Enabling new linters and checks

### DISCUSSION THREAD

**Discussion by oscr** [kubernetes-sigs/cluster-api#7170] [General]
There seems to be both a number of [new linters available in golangci-lint](https://golangci-lint.run/usage/linters/) and additional checks that can be enabled in existing metalinters such as [govet](https://golangci-lint.run/usage/linters/#govet).

I would like to suggest looking at and possibly adding some of the "bug" focused linters. Improving correctness is great and given that the signal to noise ratio isn't too bad, they might be worth having.

**Comment by sbueringer** [comment]
I think in general that's fine but we should discuss the individual linters. We try to get the right trade-off between having the right linters and trying to not set the bar to high for new contributors. Feel free to open PRs to suggest new linters.

**Reply by fabriziopandini** [reply]
+1 to do this step by step

**Comment by fabriziopandini** [comment]
This is being done step by step as suggested above since some time now
